### PR TITLE
GH-65046: Fix docs about logging in `asyncio`

### LINKED ISSUE #65046: asyncio docs should call out that network logging is a no-no
BPO | [20847](https://bugs.python.org/issue20847)
--- | :---
Nosy | @vstinner, @merwok, @socketpair, @1st1, @Mariatta

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/vstinner'
closed_at = None
created_at = <Date 2014-03-03.20:12:20.295>
labels = ['type-feature', 'docs', 'expert-asyncio']
title = 'asyncio docs should call out that network logging is a no-no'
updated_at = <Date 2016-11-02.21:28:11.952>
user = 'https://github.com/gvanrossum'
```

bugs.python.org fields:
```python
activity = <Date 2016-11-02.21:28:11.952>
actor = 'Mariatta'
assignee = 'vstinner'
closed = False
closed_date = None
closer = None
components = ['Documentation', 'asyncio']
creation = <Date 2014-03-03.20:12:20.295>
creator = 'gvanrossum'
dependencies = []
files = []
hgrepos = []
issue_num = 20847
keywords = ['patch']
message_count = 7.0
messages = ['212664', '212666', '212671', '212689', '279181', '279184', '279371']
nosy_count = 6.0
nosy_names = ['vstinner', 'eric.araujo', 'socketpair', 'yselivanov', 'Winterflower', 'Mariatta']
pr_nums = []
priority = 'normal'
resolution = None
stage = 'needs patch'
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue20847'
versions = ['Python 3.4']
```

</p></details>

### PR DESCRIPTION
**Pull request by kumaraditya303**
* Issue: gh-65046

Closes https://github.com/python/cpython/issues/65046

### CODE REVIEW

File: Doc/library/asyncio-dev.rst [hunk 1]
```diff
@@ -148,6 +148,10 @@ adjusted::
    logging.getLogger("asyncio").setLevel(logging.WARNING)
 
 
+Network logging can block the event loop, it is recommended to use
+a separate thread for handling logs or use non blocking io.
```
Comment (merwok) [inline Doc/library/asyncio-dev.rst]: I think it should be spelled `non-blocking I/O`, but the bigger issue is that it’s not immediately clear what one should use: after all, we are in the async I/O docs!   Is there a section in the `os` module or another part of the stdlib that explains how to do non-asyncio non-blocking I/O?

Comment (kumaraditya303) [inline Doc/library/asyncio-dev.rst]:
```suggestion
a separate thread for handling logs or use non-blocking IO.
```

File: Doc/library/asyncio-dev.rst [hunk 2]
```diff
@@ -148,6 +148,10 @@ adjusted::
    logging.getLogger("asyncio").setLevel(logging.WARNING)
 
 
+Network logging can block the event loop, it is recommended to use
```
Comment (gvanrossum) [inline Doc/library/asyncio-dev.rst]:
```suggestion
Network logging can block the event loop. It is recommended to use
```

**Review by gvanrossum — APPROVED** [review]
One nit, I don't think a comma is quite right here. I'll apply my suggestion and merge your PR.

**Comment by merwok** [comment]
No reaction to my remark about the vagueness of the new recommendation?

**Comment by gvanrossum** [comment]
> No reaction to my remark about the vagueness of the new recommendation?

I apologize, I was in "get things merged" mode and missed that.

That said, I doubt we have something more specific than "use threads or write to disk" -- and while writing to disk is technically not perfect, it's generally considered good enough for asyncio (we've been told that there are async APIs for writing to disk on some platforms, and IIRC there are 3rd party packages that wrap them, but the benefits are at best marginal, and the code isn't portable.

Also, I think the warning is about configuring a logger that uses network I/O rather than about how user code should do their own logging. The point is that asyncio _itself_ sometimes calls the logging module and the user should avoid messing that up by accidentally configuring a network-using logger.

User code can just use the run_in_executor API to do the logging from a thread, but we don't want to do that for asyncio's own logging.

**Pull request merged**